core/src/main/scala/kafka/admin/ConsumerGroupCommand.scala

```diff
@@ -55,83 +57,137 @@ object ConsumerGroupCommand {
+        new ZkConsumerGroupService(opts, consumerGroupOutputWriter)
       } else {
+        println("Note: This will only show information about consumers that use the Java consumer API (non-ZooKeeper-based consumers).\n")
         new KafkaConsumerGroupService(opts)
       }
     }
 
     try {
       if (opts.options.has(opts.listOpt))
-        consumerGroupService.list()
-      else if (opts.options.has(opts.describeOpt))
-        consumerGroupService.describe()
+        consumerGroupService.listGroups().foreach { consumerGroupOutputWriter.printMessage(_) }
+      else if (opts.options.has(opts.describeOpt)) {
+        consumerGroupService.describeGroup() match {
+          case None =>
+            consumerGroupOutputWriter.printError(s"The consumer group '${opts.options.valuesOf(opts.groupOpt).asScala.head}' does not exist.")
+          case Some(assignments) =>
+            if (assignments.isEmpty)
+              consumerGroupOutputWriter.printError(s"Consumer group '${opts.options.valuesOf(opts.groupOpt).asScala.head}' is rebalancing.")
+            else
+              consumerGroupOutputWriter.printAssignment(assignments)
+        }
+      }
       else if (opts.options.has(opts.deleteOpt)) {
         consumerGroupService match {
-          case service: ZkConsumerGroupService => service.delete()
-          case _ => throw new IllegalStateException(s"delete is not supported for $consumerGroupService")
+          case service: ZkConsumerGroupService => service.deleteGroups()
+          case _ => throw new IllegalStateException(s"delete is not supported for $consumerGroupService.")
         }
       }
     } catch {
       case e: Throwable =>
-        println("Error while executing consumer group command " + e.getMessage)
-        println(Utils.stackTrace(e))
+        consumerGroupOutputWriter.printError("Executing consumer group command failed due to " + e.getMessage, Some(e))
     } finally {
       consumerGroupService.close()
     }
   }
 
-  sealed trait ConsumerGroupService {
+  case class PartitionAssignmentState(group: String, coordinator: Option[Node], topic: Option[String], partition: Option[Int],
+                                      offset: Option[Long], lag: Option[Long], memberId: Option[String],
+                                      memberHost: Option[String], clientId: Option[String], logEndOffset: Option[Long])
 
-    def list(): Unit
+  class ConsumerGroupOutputWriter extends Logging {
 
-    def describe() {
-      describeGroup(opts.options.valueOf(opts.groupOpt))
+    def printError(msg: String, e: Option[Throwable] = None): Unit = {
+      println("Error: " + msg)
+      if (e.isDefined)
+        e.map(debug("Exception in consumer group command", _))
     }
 
-    def close(): Unit
+    def printMessage(msg: String): Unit = println(msg)
 
-    protected def opts: ConsumerGroupCommandOptions
+    def printAssignment(groupAssignment: Array[PartitionAssignmentState]): Unit = {
```

Comment:
We could, but we would need to pass more arguments now that we want to distinguish between empty group and rebalancing group. Plus, in the case of an empty assignment we are using the `printError` method, instead of actually printing assignments. Do you still think we should change it?